src/EventHandlers/CLPool.ts

```diff
@@ -10,12 +10,69 @@ import {
   CLPool_SetFeeProtocol,
   CLPool_Swap,
   CLPoolAggregator,
+  Token,
 } from "generated";
 import { set_whitelisted_prices } from "../PriceOracle";
 import { normalizeTokenAmountTo1e18 } from "../Helpers";
 import { multiplyBase1e18, abs } from "../Maths";
 import { updateCLPoolAggregator } from "../Aggregators/CLPoolAggregator";
 
+/**
+ * Updates the fee amounts for a CLPoolAggregator based on event data.
+ * 
+ * This function calculates the new total fees for both tokens in a liquidity pool
+ * and their equivalent value in USD. It normalizes the token amounts to a base of 1e18
+ * for consistent calculations and updates the total fees in the aggregator.
+ * 
+ * @param clPoolAggregator - The current state of the CLPoolAggregator, containing existing fee data.
+ * @param event - The event data containing the fee amounts for token0 and token1.
+ * @param token0Instance - The instance of token0, containing its decimals and price per USD.
+ * @param token1Instance - The instance of token1, containing its decimals and price per USD.
+ * 
+ * @returns An object containing the updated total fees for token0, token1, and their equivalent in USD.
+ * 
+ * The returned object has the following structure:
+ * - `totalFees0`: The updated total fees for token0, normalized to 1e18.
+ * - `totalFees1`: The updated total fees for token1, normalized to 1e18.
+ * - `totalFeesUSD`: The updated total fees in USD, calculated using the normalized token fees and their prices.
+ */
+function updateCLPoolFees(
```

Comment:
Code is shared between the collect and collectFees handler, so I just abstracted this out.